### ISSUE TITLE
Unable to connect to Elasticsearch at http://elasticsearch:9200

### ISSUE THREAD

**Issue by ajeetraina** [ahromis/swarm-elk#3]
I tried setting up ELK stack on Google Cloud Platform but looks like your docker-compose file has the below entry hard-coded:

```kibana:
    image: kibana:latest
    ports:
      - "5601:5601"
    environment:
      ELASTICSEARCH_URL: http://elasticsearch:9200
    networks:
      - elk
    deploy:
      mode: replicated
      replicas: 1
    healthcheck:
      test: wget -qO- http://localhost:5601 > /dev/null
      interval: 30s
      retries: 3
```

Due to which it tried to reach out to hostname called elasticsearch and fails out there.

**Comment by ahromis** [comment]
Hi @ajeetraina , this the expect behavior of Docker Swarm mode, as the elasticsearch service should register in DNS as the service name.

Is this what you aren't seeing on GCP? What Docker engine are you running? Which orchestrator? This information will help me in trying to duplicate the issue. Thanks.

**Comment by ajeetraina** [comment]
Hi,

I am seeing it under GCP. I am running the latest 17.06 swarm mode on
Ubuntu 16.04 instance.

Ajeet




On 24-Jul-2017 7:10 AM, "Andrew Hromis" <notifications@github.com> wrote:

Hi @ajeetraina <https://github.com/ajeetraina> , this the expect behavior
of Docker Swarm mode, as the elasticsearch service should register in DNS
as the service name.

Is this what you aren't seeing on GCP? What Docker engine are you running?
Which orchestrator? This information will help me in trying to duplicate
the issue. Thanks.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/ahromis/swarm-elk/issues/3#issuecomment-317299165>, or mute
the thread

**Comment by ajeetraina** [comment]
I understand that the service discovery comes into picture and hence the it should be able to locate elasticsearch by name. Even though I see it is throwing error saying unable to reach out to http://elasticsearch

**Comment by ahromis** [comment]
Can you launch another container on the same network then exec into the container and try to resolve `elasticsearch`? Or you can use this container for troubleshooting network issues such as this: https://github.com/nicolaka/netshoot#drill

If it registers in DNS then I'm guessing it might be related to timing, and Elasticsearch could need more time to start up before Kibana comes up.

**Comment by ahromis** [comment]
@ajeetraina this might have been related to the healtcheck, I just closed out issue #4. Can you try again with the latest from master? I think Elasticsearch might have not been staying up and healthy for long enough and Kibana was complaining about it.

**Comment by ahromis** [comment]
Closing issue, I'm quite certain this was tied to the healthcheck in the previous comment.
